thumbnails/thumbnails.go

```diff
@@ -218,15 +235,23 @@ func processSegment(input string, thumbOut string) error {
 
 var segmentPrefix = []string{"index", "clip_"}
 
-func thumbFilename(segmentURI string) (string, error) {
+func segmentIndex(segmentURI string) (int64, error) {
 	// segmentURI will be indexX.ts or clip_X.ts
 	for _, prefix := range segmentPrefix {
 		segmentURI = strings.TrimPrefix(segmentURI, prefix)
 	}
 	index := strings.TrimSuffix(segmentURI, ".ts")
-	i, err := strconv.ParseInt(index, 10, 32)
+	i, err := strconv.ParseInt(index, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("thumbFilename failed for %s: %w", segmentURI, err)
+	}
+	return i, nil
+}
+
+func thumbFilename(segmentURI string, segmentOffset int64) (string, error) {
+	i, err := segmentIndex(segmentURI)
 	if err != nil {
-		return "", fmt.Errorf("thumbFilename failed for %s: %w", segmentURI, err)
+		return "", err
 	}
-	return fmt.Sprintf("keyframes_%d.jpg", i), nil
+	return fmt.Sprintf("keyframes_%d.jpg", i-segmentOffset), nil
```

Comment:
Doesn't it complicate things that now the segments will have different numbering than thumbnails? 🤔

Reply:
That's what I thought initially but then I realised that the output manifests do start from zero, it's only the input manifest which doesn't